src/simulators/stabilizer/stabilizer_state.hpp

```diff
@@ -536,10 +537,29 @@ void State::apply_set_stabilizer(const Clifford::Clifford &clifford) {
 
 void State::apply_save_stabilizer(const Operations::Op &op,
                                 ExperimentResult &result) {
-  std::string key = (op.string_params[0] == "_method_") ? "stabilizer" : op.string_params[0];
+  std::string key = op.string_params[0];
+  OpType op_type = op.type;
+  switch (op_type) {
+    case OpType::save_clifford: {
+      if (key == "_method_") {
+        key = "clifford";
+      }
+      break;
+    }
+    case OpType::save_stabilizer:
+    case OpType::save_state: {
+      if (key == "_method_") {
+        key = "stabilizer";
+      }
+      op_type = OpType::save_stabilizer;
+      break;
+    }
+    default:
+      // We shouldn't ever reach here...
+      throw std::invalid_argument("Invalid save state instruction for stabilizer");
+  }
```

Comment:
I'm happy!

(I think the braces in `case` are superfluous, but it shouldn't matter here.)